AEPMessaging/Sources/Messaging.swift

```diff
@@ -306,16 +394,141 @@ public class Messaging: NSObject, Extension {
             return
         }
     }
-    
+
+    func propositionInfoForMessageId(_ messageId: String) -> PropositionInfo? {
+        propositionInfo[messageId]
+    }
+
+    // swiftlint:disable function_body_length
+    func parsePropositions(_ propositions: [PropositionPayload]?, expectedSurfaces: [String], clearExisting: Bool, persistChanges: Bool = true) -> [LaunchRule] {
```

Comment:
feels like we are shoving too much logic into this method (and to an extent `handleEdgePersonalizationNotification` as well) when we're handling both IAM and feeds.  do you think the time spent would be justified if we clean it up and separate the logic out into multiple methods?

Reply:
Yeah I plan to refactor this code a bit more.